### ISSUE TITLE
NFT minting can be blocked by various limits

### ISSUE THREAD

**Issue by dmtrjsg** [Joystream/atlas#2823]
Details covered here: 

* https://github.com/Joystream/joystream/pull/3609

⚠️ Blocked by - https://github.com/Joystream/joystream/issues/3868

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1202132419573087/1202410475193821) by [Unito](https://www.unito.io)

**Comment by dmtrjsg** [comment]
Linked up with corresponding QN issue

**Comment by bedeho** [comment]
Out of scope for `Carthage` due to freeze here https://github.com/Joystream/joystream/issues/3882, so removing label.
